### PR TITLE
[go] Update to esbuild script

### PR DESCRIPTION
**Pull request by cb1kenobi**
Updated the `go` build script to use the new esbuilt script. Moved the `.ts` code into a `src` directory.

### CODE REVIEW

File: packages/go/tsconfig.json [hunk 1]
```diff
@@ -11,8 +11,11 @@
...
     "noUnusedParameters": true,
     "noImplicitThis": false,
+    "outDir": "./dist",
     "types": ["node", "jest"],
     "strict": true,
     "target": "ES2021"
-  }
+  },
+  "include": ["src/**/*"],
+  "exclude": ["node_modules"]
```
Comment (TooTallNate) [inline packages/go/tsconfig.json]: `node_modules` is excluded by default

```suggestion
  "include": ["src/**/*"]
```

File: .changeset/dry-baboons-happen.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
Comment (styfle) [inline .changeset/dry-baboons-happen.md:3]: Should be patch so it gets published

File: packages/go/build.mjs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { esbuild } from '../../utils/build.mjs';
+
+await esbuild({ bundle: true, external: ['@vercel/build-utils'] });
```
Comment (styfle) [inline packages/go/build.mjs]: Should also mark `dependencies` as external

**Review by TooTallNate — APPROVED** [review]

File: .changeset/dry-baboons-happen.md (hunk 2 above)
Comment (cb1kenobi) [reply .changeset/dry-baboons-happen.md:3]: Fixed

File: packages/go/build.mjs (hunk 3 above)
Comment (cb1kenobi) [reply packages/go/build.mjs]: Non issue now

**Review by styfle — APPROVED** [review]

**Pull request merged**